C++/Cpp-Templates-Second-Edition/basics/include/basics/foreach.hpp

```diff
@@ -0,0 +1,16 @@
+#ifndef FOREACH_HPP
+#define FOREACH_HPP
```

Comment:
While #pragma once is not a part of the language it is still supported by many compilers so you could think about using it instead old guards.